app/grandchallenge/core/utils/markdown.py

```diff
@@ -1,7 +1,39 @@
+from xml.etree import ElementTree
+
+from bs4 import BeautifulSoup
 from markdown import Extension
 from markdown.treeprocessors import Treeprocessor
 
 
+class HtmlElementWrapper:
+    def __init__(self, element):
+        self.element = element
+
+    def set_css_class(self, class_name: str):
+
+        if isinstance(self.element, ElementTree.Element):
+
+            current_class = self.element.attrib.get("class", "")
+
+            if class_name not in current_class:
+                new_class = f"{current_class} {class_name}".strip()
+                self.element.set("class", new_class)
+
+        elif isinstance(self.element, BeautifulSoup().new_tag("").__class__):
+
+            if "class" not in self.element.attrs:
+                self.element.attrs["class"] = []
+
+            current_class = self.element["class"]
+
+            for name in class_name.split(" "):
+                if class_name not in current_class:
+                    current_class.append(name)
+
+    def get_element(self):
```

Comment:
You shouldn't need a `get_` method for public attributes. Why is this thing a class at all? Seems like it should just be a function `set_css_class` which takes an element and class name and returns an element.

Reply:
A `staticmethod` in `BS4Treeprocessor` would make the most sense. Other things:

- it should be kwargs only
- The fallthrough in the switch statement needs to be explicitly handled

Reply:
Do you mean by "kwargs only" that the method has a signature like 
```
@staticmethod
set_css_class(**kwargs)
``` 
or that it uses input from arguments only?
or to imply that `self` is not needed in the static method?

Reply:
I made a commit, not sure if it is exactly as what you have in mind.